src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/client/CaseWorkerApiClient.java

```diff
@@ -147,9 +116,70 @@ public RequestSpecification getMultipleAuthHeaders(String userToken) {
-        return mapper.readTree(jsonString);
+    public List<CaseWorkersProfileCreationRequest> updateCaseWorkerProfileRequest(String... email) {
+        List<CaseWorkerLocationRequest> locationRequestList = List.of(
+                CaseWorkerLocationRequest
+                        .caseWorkersLocationRequest()
+                        .location("updated location")
+                        .locationId(3)
+                        .isPrimaryFlag(true).build(),
+                CaseWorkerLocationRequest
+                        .caseWorkersLocationRequest()
+                        .location("updated added new location")
+                        .locationId(1).isPrimaryFlag(false).build());
+
+        List<CaseWorkerRoleRequest> roleRequests = List.of(
+                CaseWorkerRoleRequest
+                        .caseWorkerRoleRequest()
+                        .role("senior-tribunal-caseworker")
+                        .isPrimaryFlag(true).build(),
+                CaseWorkerRoleRequest
+                        .caseWorkerRoleRequest()
+                        .role("tribunal-caseworker")
+                        .isPrimaryFlag(false).build());
+
+
+        List<CaseWorkerWorkAreaRequest> areaRequests = List.of(
+                CaseWorkerWorkAreaRequest
+                        .caseWorkerWorkAreaRequest()
+                        .serviceCode("BAA1")
+                        .areaOfWork("Non-Money Claims").build(),
+                CaseWorkerWorkAreaRequest
+                        .caseWorkerWorkAreaRequest()
+                        .serviceCode("BAA9")
+                        .areaOfWork("Possession Claims").build());
+
+        Set<String> idamRoles = new HashSet<>();
+
+        String emailToUsed = isNotEmpty(email) && nonNull(email[0]) ? email[0] : generateRandomEmail();
+        setEmailsTobeDeleted(emailToUsed);
+        return ImmutableList.of(
+                CaseWorkersProfileCreationRequest
+                        .caseWorkersProfileCreationRequest()
+                        .firstName("updatedFirstName")
+                        .lastName("updatedLastName")
+                        .emailId(emailToUsed)
+                        .regionId(2)
+                        .region("County")
+                        .userType("CTRT")
+                        .suspended(false)
+                        .idamRoles(idamRoles)
+                        .baseLocations(locationRequestList)
+                        .roles(roleRequests)
+                        .workerWorkAreaRequests(areaRequests).build());
     }
 
+    public Response createUserProfiles(List<CaseWorkersProfileCreationRequest> caseWorkersProfileCreationRequests) {
+        Response response = getMultipleAuthHeadersInternal()
+                .body(caseWorkersProfileCreationRequests)
+                .post("/refdata/case-worker/users/")
+                .andReturn();
+        log.info(":: Create user profile response status code :: " + response.statusCode());
```

Comment:
do we need logs in func test cases?

Reply:
just to see what was the response status code. This is useful whenever the functional test cases fails on AAT